app/stac_api/serializers.py

```diff
@@ -84,6 +86,36 @@ def update_or_create_links(model, instance, instance_type, links_data):
     )
 
 
+class CommonJsonValidatorSerializer(serializers.ModelSerializer):
+
+    def validate(self, attrs):
+
+        # see: https://www.django-rest-framework.org/api-guide/serializers/#object-level-validation
+
+        # common validation part for collection, item and asset:
+        expected_payload = list(self.fields.keys())
+        expected_payload_read_only = [
+            field for field in self.fields if self.fields[field].read_only
+        ]
+
+        for key in self.initial_data.keys():
+            if key not in expected_payload:
+                raise ValidationError(detail=_("Unexpected property in payload"), code=key)
+            if key in expected_payload_read_only:
+                raise ValidationError(detail=_("Found read-only property in payload"), code=key)
+
+        # add specific validation for ItemSerializer
+        if self.__class__.__name__ == "ItemSerializer":
+            validate_item_properties_datetimes(
+                attrs.get('properties_datetime', None),
+                attrs.get('properties_start_datetime', None),
+                attrs.get('properties_end_datetime', None),
+                partial=self.partial
+            )
+
+        return attrs
```

Comment:
I don't really like the uses of inherithance here, and if used it should be renamed as mixin, as it follow the mixin pattern. I would keep it simple by simply overriding the `validate()` method of the three main serializer (Collection, Item and Asset) which would call the validator function by passing `self.initial_data`

Reply:
No inheritance here anymore, wrote three separate `validate()` now